packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-checkbox-group/gux-form-field-checkbox-group.service.ts

```diff
@@ -0,0 +1,56 @@
+import simulateNativeEvent from '@utils/dom/simulate-native-event';
+
+export function setAllCheckboxInputs(
+  root: HTMLElement,
+  checked: boolean
+): void {
+  getNestedCheckboxInputs(root).forEach(checkboxInput => {
+    if (checkboxInput.checked !== checked && !checkboxInput.disabled) {
+      checkboxInput.checked = checked;
+      simulateNativeEvent(checkboxInput, 'input');
+      simulateNativeEvent(checkboxInput, 'change');
+    }
+  });
+}
+
+export function setParentCheckboxElementCheckedState(
+  root: HTMLElement,
+  mainCheckboxElement: HTMLInputElement
+): void {
+  if (mainCheckboxElement) {
+    const { checkedCheckboxes, totalCheckboxes } = getSelectedColumnCount(root);
+    if (checkedCheckboxes === 0) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = false;
+    } else if (checkedCheckboxes === totalCheckboxes) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = true;
+    } else {
+      mainCheckboxElement.indeterminate = true;
+    }
+  }
+}
+
+export function getSelectedColumnCount(root: HTMLElement): {
+  checkedCheckboxes: number;
+  totalCheckboxes: number;
+} {
+  const totalCheckboxes = getNestedCheckboxInputs(root).length;
+  const checkedCheckboxes = getCheckedCheckboxInputs(root).length;
```

Comment:
What are your thoughts on not having a separate function to get checked checkbox inputs and putting it inline here like so. I feel like this way is more functional.

```
const checkedCheckboxes = getNestedCheckboxInputs(root).filter(x => x.checked).length;
return { checkedCheckboxes, totalCheckboxes };
```

Reply:
I think it is fine as is. Your suggestion is good as well but I think I would like to just get this PR merged instead of doing more refactors.